readyapi4j-groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/pro/DataGenDataSourceTestStepDslTest.groovy

```diff
@@ -0,0 +1,186 @@
+        assert stringDataGenerator.propertyName == 'Strings'
+        assert stringDataGenerator.minimumCharacters == 8
+        assert stringDataGenerator.maximumCharacters == 20
+        assert !stringDataGenerator.useSpaces
+        assert !stringDataGenerator.useDigits
+        assert !stringDataGenerator.usePunctuationMarks
+        assert stringDataGenerator.useLetters
+    }
+
+    @Test
+    void createsRecipeWithDataSourceTestStepWithCustomStringDataGen() throws Exception {
+        TestRecipe testRecipe = recipe {
+            dataGenDataSource 'CustomStringDataGenDataSource', {
+                customString 'ConstantString', 'This is a custom string!'
+            }
+        }
+        DataGenDataSource dataSource = extractDataGenDataSource(testRecipe)
+        CustomStringDataGenerator customStringDataGenerator = dataSource.dataGenerators[0] as CustomStringDataGenerator
+        assert customStringDataGenerator.propertyName == 'ConstantString'
+        assert customStringDataGenerator.value == 'This is a custom string!'
+    }
+
+    @Test
+    void createsRecipeWithDataSourceTestStepWithShortStateNameDataGen() throws Exception {
+        TestRecipe testRecipe = recipe {
+            dataGenDataSource 'ShortStateNamesDataGenDataSource', {
+                abbreviatedStateNames 'StateNames'
+            }
+        }
+        DataGenDataSource dataSource = extractDataGenDataSource(testRecipe)
+        StateNameDataGenerator stateNameDataGenerator = dataSource.dataGenerators[0] as StateNameDataGenerator
+        assert stateNameDataGenerator.propertyName == 'StateNames'
+        assert stateNameDataGenerator.nameFormat == StateNameDataGenerator.NameFormatEnum.ABBREVIATED
+    }
+
+    @Test
+    void createsRecipeWithDataSourceTestStepWithFullStateNameDataGen() throws Exception {
+        TestRecipe testRecipe = recipe {
+            dataGenDataSource 'FullStateNamesDataGenDataSource', {
+                fullStateNames 'FullStateNames'
+            }
+        }
+        DataGenDataSource dataSource = extractDataGenDataSource(testRecipe)
+        StateNameDataGenerator stateNameDataGenerator = dataSource.dataGenerators[0] as StateNameDataGenerator
+        assert stateNameDataGenerator.propertyName == 'FullStateNames'
+        assert stateNameDataGenerator.nameFormat == StateNameDataGenerator.NameFormatEnum.FULL
+    }
+
+    @Test
+    void createsRecipeWithDataSourceTestStepWithDateTimeDataGen() throws Exception {
+        Date startDate = Date.parse("yyyy-MM-dd hh:mm:ss", "2014-04-03 1:23:45")
+        Date endDate = Date.parse("yyyy-MM-dd hh:mm:ss", "2016-04-03 23:59:59")
+
+        TestRecipe testRecipe = recipe {
+            dataGenDataSource 'FullStateNamesDataGenDataSource', {
+                dateAndTimeValues 'DateAndTimes', {
+                    startingAt startDate
+                    endingAt endDate
+                    formatISO_8601
```

Comment:
Here, i think I'd prefer a setter:

format = ISO_8601 (we could have an enum with supported formats)
or
 format = 'ISO_8601'